tests/test_request/test_threadPool.py

```diff
@@ -0,0 +1,122 @@
+import threading
+import time
+import random
+import itertools
+
+from unittest import mock
+
+import pytest
+
+from lazyflow.request.threadPool import ThreadPool
+
+
+NUM_WORKERS = 4
+
+
+class Task:
+    _counter = itertools.count()
+
+    def __init__(self, fn):
+        self.fn = fn
+        self.assigned_worker = None
+        self.priority = next(self._counter)
+
+    def __lt__(self, other):
+        return self.priority > self.priority
+
+    def __call__(self):
+        self.fn()
+
+
+@pytest.fixture
+def pool():
+    return ThreadPool(NUM_WORKERS)
+
+
+def test_thread_pool_starts_workers(pool: ThreadPool):
+    assert pool.num_workers == NUM_WORKERS
+    assert len(pool.workers) == NUM_WORKERS
+
+    for w in pool.workers:
```

Comment:
```python
assert all(isinstance(w, threading.Thread) for w in pool.workers))
```

Reply:
There is a problem to consider with failing test:
```
E        +  where False = all(<generator object test_foo.<locals>.<genexpr> at 0x7ff77d0ca7c8>)
```
While current form shows exact offending object without further investigation